wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WikibaseDataEditor.java

```diff
@@ -551,10 +556,84 @@ public ItemDocument updateTermsStatements(ItemDocument currentDocument,
 		TermStatementUpdate termStatementUpdate = new TermStatementUpdate(currentDocument,
 				addStatements, deleteStatements,
 				addLabels, addDescriptions, addAliases, deleteAliases);
-
-		return (ItemDocument) this.wbEditEntityAction.wbEditEntity(currentDocument
+		
+		if (termStatementUpdate.isNull()) {
+			return currentDocument;
+		} else {
+			return (ItemDocument) this.wbEditEntityAction.wbEditEntity(currentDocument
 				.getEntityId().getId(), null, null, null, termStatementUpdate
 				.getJsonUpdateString(), false, this.editAsBot, currentDocument
 				.getRevisionId(), summary);
+		}
+	}
+	
+	/**
+	 * Performs a null edit on an item. This has some effects on Wikibase,
+	 * such as refreshing the labels of the referred items in the UI.
+	 * 
+	 * @param currentDocument
+	 * 			the document to perform a null edit on
+	 * @param summary
+	 * 			a summary that can be passed to Wikibase (although it is not
```

Comment:
If it is a real null edit the summary is ignored because no new revision is created in the database. I would be in favor of removing this argument.

Reply:
great, I wasn't sure about this, so let's do that.